pkg/util/ldaputil/parseFilter.go

```diff
@@ -0,0 +1,35 @@
+package ldaputil
+
+import (
+	"bytes"
+	"strings"
+	"text/template"
+
+	"github.com/go-ldap/ldap/v3"
+)
+
+func ParseFilter(filter string, username string) (string, error) {
```

Comment:
First of all, who uses this function?

Second, the testcase commit should be merged into this commit. An implementation with a testcase is considered as atomic commit. Do not separate them.